src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPool2.cs

```diff
@@ -99,8 +96,11 @@ private void AllocateSlab()
             var basePtr = slab.ArrayPtr;
             var firstOffset = (int)((_blockStride - 1) - ((ulong)(basePtr + _blockStride - 1) % _blockStride));
 
-            for (var offset = firstOffset;
-                offset + _blockLength <= _slabLength;
+            var poolAllocationLength = _slabLength - (_blockLength + _blockStride);
```

Comment:
I think semantically it makes more sense for `poolAllocationLength` to be `_slabLength - _blockStride`. The for condition then could go back to `offset + _blockLength < poolAllocationLength`.